src/storage/compression/compression_float.cpp

```diff
@@ -0,0 +1,240 @@
+#include "storage/compression/compression_float.h"
+
+#include "alp/encode.hpp"
+#include "common/constants.h"
+#include "common/utils.h"
+#include <ranges>
+
+namespace kuzu {
+namespace storage {
+
+namespace {
+static constexpr common::idx_t BITPACKING_CHILD_IDX = 0;
+
+template<std::floating_point T>
+common::LogicalType getBitpackingLogicalType() {
+    if constexpr (std::is_same_v<T, float>) {
+        return common::LogicalType::INT32();
+    } else {
+        return common::LogicalType::INT64();
+    }
+}
+} // namespace
+
+template<std::floating_point T>
+size_t EncodeException<T>::numPagesFromExceptions(size_t exceptionCount) {
+    return common::ceilDiv(static_cast<uint64_t>(exceptionCount),
+        common::BufferPoolConstants::PAGE_4KB_SIZE / sizeBytes());
+}
+
+template<std::floating_point T>
+size_t EncodeException<T>::exceptionBytesPerPage() {
+    return common::BufferPoolConstants::PAGE_4KB_SIZE / sizeBytes() * sizeBytes();
+}
+
+template<std::floating_point T>
+bool EncodeException<T>::operator<(const EncodeException<T>& o) const {
+    return posInChunk < o.posInChunk;
+}
+
+template<std::floating_point T>
+EncodeException<T> EncodeExceptionView<T>::getValue() const {
+    EncodeException<T> ret;
+    std::memcpy(&ret.value, bytes, sizeof(ret.value));
+    std::memcpy(&ret.posInChunk, bytes + sizeof(ret.value), sizeof(ret.posInChunk));
+    return ret;
+}
+
+template<std::floating_point T>
+void EncodeExceptionView<T>::setValue(EncodeException<T> exception) {
+    std::memcpy(bytes, &exception.value, sizeof(exception.value));
+    std::memcpy(bytes + sizeof(exception.value), &exception.posInChunk,
+        sizeof(exception.posInChunk));
+}
+
+template<std::floating_point T>
+uint64_t FloatCompression<T>::compressNextPage(const uint8_t*&, uint64_t, uint8_t*, uint64_t,
+    const struct CompressionMetadata&) const {
+    KU_UNREACHABLE;
+}
+
+template<std::floating_point T>
+uint64_t FloatCompression<T>::compressNextPageWithExceptions(const uint8_t*& srcBuffer,
+    uint64_t srcOffset, uint64_t numValuesRemaining, uint8_t* dstBuffer, uint64_t dstBufferSize,
+    uint8_t* exceptionBuffer, [[maybe_unused]] uint64_t exceptionBufferSize,
+    uint64_t& exceptionCount, const struct CompressionMetadata& metadata) const {
+
+    if (metadata.compression == CompressionType::UNCOMPRESSED) {
+        return Uncompressed(sizeof(T)).compressNextPage(srcBuffer, numValuesRemaining, dstBuffer,
+            dstBufferSize, metadata);
+    }
```

Comment:
this should be removeable, replace with `KU_ASSERT(metadata.compression == CompressionType::FLOAT)`